scripts/bootstrapTypes/genTypeExports.ts

```diff
@@ -0,0 +1,147 @@
+			types.add(node.typeName.getText(sourceFile));
+		}
+		ts.forEachChild(node, visit);
+	}
+	visit(sourceFile);
+	return types;
+}
+
+const componentsProps: [string, string, number][] = [];
+
+const components = await readdir(coreBootstrapComponents);
+for (const component of components) {
+	const sourceFile = program.getSourceFile(path.join(coreBootstrapComponents, `${component}/${component}.ts`))!;
+	const sourceFileSymbol = typeChecker.getSymbolAtLocation(sourceFile)!;
+
+	// The first stage of the generator is determining what we need to export.
+	// We take all type aliases or interface exports from the file
+	// core-bootstrap/src/components/${component}/${component}.ts and filter out Extra or Common ones.
+	const exportsList = typeChecker.getExportsOfModule(sourceFileSymbol);
+	let exports = '';
+	const exportNames = new Set();
+	for (const bootstrapExport of exportsList.filter(
+		(btsExport) => !btsExport.name.endsWith('CommonPropsAndState') && !btsExport.name.endsWith('ExtraProps') && !btsExport.name.startsWith('Common'),
+	)) {
+		exportNames.add(bootstrapExport.name);
+		const node = bootstrapExport.getDeclarations()![0];
+		if (ts.isTypeAliasDeclaration(node)) {
+			// type aliases are copied as defined
+			exports += node.getText() + '\n\n';
+		} else if (ts.isInterfaceDeclaration(node)) {
+			// interfaces are resolved to obtain a *flattened* version
+			exports += `export interface ${bootstrapExport.name}${node.typeParameters?.length ? `<${node.typeParameters.map((typeParam) => typeParam.getText()).join(', ')}>` : ''} {\n`;
+			for (const property of typeChecker.getTypeAtLocation(node).getProperties()) {
+				const propertyDeclaration = property.getDeclarations()![0];
+				// TODO the js documentation is copied as-is, though it could be interesting for Props interfaces
+				// to add the tag @defaultValue based on the getDefaultConfig functions
+				exports += `\t/**\n\t * ${ts.displayPartsToString(property.getDocumentationComment(typeChecker)).split('\n').join('\n\t * ')}\n\t */\n`;
+				exports += `\t${propertyDeclaration.getText()}\n`;
+			}
+			exports += `}\n\n`;
+			if (bootstrapExport.name === `${component.slice(0, 1).toUpperCase()}${component.slice(1)}Props`) {
+				componentsProps.push([component, bootstrapExport.name, node.typeParameters?.length ?? 0]);
+			}
+		}
+	}
+	const allUsedTypes = getAllTypesFromSourceCode(exports);
+
+	// We now resolve all necessary imports to add on top of the generated file.
+	// For this we go through all files in core-bootstrap and core to resolve type-only imports by name / location.
+	// Once this map is built, we change the `@agnos-ui/core` location to be `@agnos-ui/${framework}-headless` instead.
+	const mapImportsByModule = new Map<string, Set<string>>();
+	const visited = new Set();
+	const queue: string[] = [`@agnos-ui/core-bootstrap/components/${component}/${component}`];
+	visited.add(queue[0]);
+	visited.add('@agnos-ui/core/components/commonProps');
+
+	while (queue.length) {
```

Comment:
Rather than re-implementing an exploration of files to find types as you do here, I would suggest using typescript to resolve types, cf [my commit here](https://github.com/divdavem/AgnosUI/commit/wipGenerateType)
What do you think?

Reply:
Indeed, thanks for the suggestion !